Maps_Settings.php

```diff
@@ -301,5 +301,60 @@
 		// user does not provide one.
 		$GLOBALS['egMapsLeafletZoom'] = 14;
 
+		// String. The default layer for Leaflet. This value will only be
+		// used when the user does not provide one.
+		$GLOBALS['egMapsLeafletLayer'] = 'OpenStreetMap';
+
+		$GLOBALS['egMapsLeafletOverlayLayers'] = [
+
+		];
+
+		// The definitions for the layers that should be available for the user.
+		$GLOBALS['egMapsLeafletAvailableLayers'] = [
+			'OpenStreetMap',
+			'OpenStreetMap.DE',
+			'OpenStreetMap.BlackAndWhite',
+			'OpenStreetMap.HOT',
+			'Thunderforest.OpenCycleMap',
+			'Thunderforest.Transport',
+			'Thunderforest.Landscape',
+			'Hydda.Full',
+			//'MapBox',
+			'Stamen.Toner',
+			'Stamen.Terrain',
+			'Stamen.Watercolor',
+			'Esri.WorldStreetMap',
+			'Esri.DeLorme',
+			'Esri.WorldTopoMap',
+			'Esri.WorldImagery',
+			'Esri.WorldTerrain',
+			'Esri.WorldShadedRelief',
+			'Esri.WorldPhysical',
+			'Esri.OceanBasemap',
+			'Esri.NatGeoWorldMap',
+			'Esri.WorldGrayCanvas',
+			'MapQuestOpen',
+		];
```

Comment:
It has occurred to me lately that while this is a sensible data structure for inside of the code, that for configuration it causes some issue: unless you're adding something, you'll need to redefine the array, or use some kind of filtering PHP code. So if the user wants to simply disable a single layer, the easiest thing for them to do is to copy the thing and remove the entry, which is kinda silly, and then also does not keep them up to date with the latest entries in new versions.

How about doing

```php
[
    'OpenStreetMap.BlackAndWhite' => true,
    'OpenStreetMap.HOT' => true,
]
```

Then they can easily disable one. You'd probably want to map it to a normal list when reading the config then, though that is really simple...

Thoughts @kghbln?